### ISSUE TITLE
Different randomized transforms applied to image and segmentation in ArrayDataset

### ISSUE THREAD

**Issue by marianabb** [Project-MONAI/MONAI#1102]
**Problem description**
As in the title, I am creating a simple ArrayDataset with RandRotate as transform. The image and the segmentation get different rotation angles, is this the expected behaviour?
I've also tested RandAffine as transform and got the same problem.

**To Reproduce**
```
transforms_train = Compose([LoadNifti(image_only=True),AddChannel(),RandRotate(range_x=90,prob=1.0,keep_size=True)])

train_dataset = ArrayDataset(img=img_paths, img_transform=transforms_train, seg=seg_paths, seg_transform=transforms_train)

check_loader = DataLoader(train_dataset, batch_size=1, shuffle=True) # MONAI DataLoader
check_data = first(check_loader)
img, seg = (check_data[0][0][0],check_data[1][0][0])

img_nib_m = nib.Nifti1Image(HR.numpy(), np.eye(4))
nib.save(img_nib_m,'./img.nii.gz')

img_nib_m = nib.Nifti1Image(LR.numpy(), np.eye(4))
nib.save(img_nib_m,'./seg.nii.gz')
```
The image will be rotated at a different angle when compared to the segmentation.

**Expected behavior**
Image and segmentation should be rotated by the same angle.

**Environment**
 - MacOS
 - Python 3.7
 - MONAI version 0.3.0
 - CUDA/cuDNN version (not used in this test)
 - GPU models and configuration (not used in this test)

**Comment by Nic-Ma** [comment]
Hi @marianabb ,

Thanks for your interest here.
I think your problem is because you used the same transform objects for both image and label, and it sets random seed at construction:
https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/compose.py#L210
Could you please test with:
```py
img_train = Compose([LoadNifti(image_only=True),AddChannel(),RandRotate(range_x=90,prob=1.0,keep_size=True)])
label_train = Compose([LoadNifti(image_only=True),AddChannel(),RandRotate(range_x=90,prob=1.0,keep_size=True)])
train_dataset = ArrayDataset(img=img_paths, img_transform=img_train, seg=seg_paths, seg_transform=label_train)
```

Thanks.

**Comment by marianabb** [comment]
Hi @Nic-Ma, 
the proposed change does make the transforms work as intended. Thank you very much!
